src/key_schedule.cpp

```diff
@@ -148,23 +148,23 @@ SecretTree::get(LeafIndex sender)
 
   // Derive down
   for (; curr > 0; --curr) {
-    auto curr_node = dirpath[curr];
-    auto left = tree_math::left(curr_node);
-    auto right = tree_math::right(curr_node, group_size);
-
-    auto& secret = secrets[curr_node.val];
-    secrets[left.val] =
-      derive_tree_secret(suite, secret, "tree", left, 0, secret_size);
-    secrets[right.val] =
-      derive_tree_secret(suite, secret, "tree", right, 0, secret_size);
+    auto curr_node = dirpath.at(curr);
+    auto left = curr_node.left();
+    auto right = curr_node.right();
+
+    auto& secret = secrets.at(curr_node);
```

Comment:
do we want to ensure against this lookup failing ?

Reply:
`.at()` throws if the key doesn't exist, so it performs the check for us.

Reply:
ah ! that's right .. thanks